### PR TITLE
kubeadm:Use kubeadmapiv1.SchemeGroupVersion.String() instead of kubeadm.k8s.i…

### LINKED ISSUE #2453: Use kubeadmapiv1.SchemeGroupVersion.String() instead of "kubeadm.k8s.io/v1beta3"
In this PR we are adding kubeadm API v1beta3.
https://github.com/kubernetes/kubernetes/pull/101129

in a number of places in the code base, mostly around unit tests we are hardcoding the string `kubeadm.k8s.io/v1beta3`.
Instead we should have this import:

```
kubeadmapiv1 "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/v1beta3"
```

and replace `kubeadm.k8s.io/v1beta3` with `kubeadmapiv1.SchemeGroupVersion.String()`.

this will allow for smaller DIFFs when sending big API update PRs to change the latest API version.

NOTE: this needs https://github.com/kubernetes/kubernetes/pull/101129 to merge first.

### PR DESCRIPTION
**Pull request by wangyysde**
Signed-off-by: wangyysde <net_use@bzhy.com>

#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
To smaller DIFFs when sending big API update PRs to change the latest API version,replaced kubeadm.k8s.io/v1beta3 with kubeadmapiv1.SchemeGroupVersion.String() .
#### Which issue(s) this PR fixes:

Fixes kubernetes/kubeadm#2453

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by fabriziopandini — COMMENTED** [review]
@wangyysde thanks for this PR!

I was wondering if we can fix also "kubeadm.k8s.io/v1beta2" constant around the codebase, like e.g.

https://github.com/fabriziopandini/kubernetes/blob/43a32c14f5827e42c1449b09aca1ab8624796e54/cmd/kubeadm/app/util/config/cluster_test.go#L50

by importing `kubeadmapiv1old "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/v1beta2"` and using `kubeadmapiv1old.SchemeGroupVersion.String()`

/ok-to-test

**Comment by neolit123** [comment]
+1 to also do kubeadmapiv1old in this pr.

**Comment by wangyysde** [comment]
@fabriziopandini  OK, I am try to use kubeadmapiv1old instead of kubeadm.k8s.io/v1beta2.

**Comment by wangyysde** [comment]
@neolit123 Thanks. I have run  ./hack/update-gofmt.sh locally.And it was exit normally.

File: cmd/kubeadm/app/cmd/config_test.go [hunk 1]
```diff
@@ -81,7 +82,7 @@ func TestImagesListRunWithCustomConfigPath(t *testing.T) {
 				constants.CurrentKubernetesVersion.String(),
 			},
 			configContents: []byte(dedent.Dedent(fmt.Sprintf(`
-				apiVersion: kubeadm.k8s.io/v1beta3
+				apiVersion: `+kubeadmapiv1.SchemeGroupVersion.String()+`
```
Comment (SataQiu) [inline cmd/kubeadm/app/cmd/config_test.go:85]: How about using `%s` ?

```go
configContents: []byte(dedent.Dedent(fmt.Sprintf(`
    apiVersion: %s
    kind: ClusterConfiguration
    kubernetesVersion: %s
`, kubeadmapiv1.SchemeGroupVersion.String(), constants.CurrentKubernetesVersion)))
```

File: cmd/kubeadm/app/cmd/config_test.go [hunk 2]
```diff
@@ -93,7 +94,7 @@ func TestImagesListRunWithCustomConfigPath(t *testing.T) {
 				"coredns",
 			},
 			configContents: []byte(dedent.Dedent(fmt.Sprintf(`
-				apiVersion: kubeadm.k8s.io/v1beta3
+				apiVersion: `+kubeadmapiv1.SchemeGroupVersion.String()+`
```
Comment (SataQiu) [inline cmd/kubeadm/app/cmd/config_test.go:97]: same as above

**Review by SataQiu — COMMENTED** [review]
Thanks @wangyysde !
SGTM, and add some comments.

File: cmd/kubeadm/app/cmd/config_test.go (hunk 2 above)
Comment (wangyysde) [reply cmd/kubeadm/app/cmd/config_test.go:97]: OK, I will do.

**Comment by wangyysde** [comment]
/test pull-kubernetes-e2e-kind-ipv6

/test pull-kubernetes-e2e-kind
--

**Review by neolit123 — COMMENTED** [review]
the PR seems to contain commits from other developers.
please remove them and only include your changes (for cmd/kubeadm) squashed into a single commit.

**Comment by neolit123** [comment]
> Which issue(s) this PR fixes:
> kubernetes/kubeadm#2453

this can be changed to:
```
fixes kubernetes/kubeadm#2453
```

**Comment by wangyysde** [comment]
@neolit123 Thanks. I have removed the other commits and modified  comment.

File: cmd/kubeadm/app/cmd/certs_test.go [hunk 3]
```diff
@@ -378,18 +378,21 @@ func TestGenCSRConfig(t *testing.T) {
...
+	var kubeadmConfig = fmt.Sprintf(`
+apiVersion: %s
 kind: InitConfiguration
 localAPIEndpoint:
   advertiseAddress: 192.0.2.1
 nodeRegistration:
   criSocket: /path/to/dockershim.sock
 ---
-apiVersion: kubeadm.k8s.io/v1beta3
+apiVersion: %s
```
Comment (fabriziopandini) [inline cmd/kubeadm/app/cmd/certs_test.go:389]: nit:
```suggestion
apiVersion: %[1]s
```
So we avoid to repeat the argument twice

File: cmd/kubeadm/app/cmd/init_test.go [hunk 4]
```diff
@@ -41,11 +42,10 @@ nodeRegistration:
     - c
     - d
 ---
-apiVersion: kubeadm.k8s.io/v1beta3
+apiVersion: %s
```
Comment (fabriziopandini) [inline cmd/kubeadm/app/cmd/init_test.go:45]: nit
```suggestion
apiVersion: %[1]s
```

File: cmd/kubeadm/app/phases/upgrade/staticpods_test.go [hunk 5]
```diff
@@ -67,26 +69,25 @@ bootstrapTokens:
 - token: ce3aa5.5ec8455bb76b379f
   ttl: 24h
 ---
-apiVersion: kubeadm.k8s.io/v1beta3
+apiVersion: %s
```
Comment (fabriziopandini) [inline cmd/kubeadm/app/phases/upgrade/staticpods_test.go:72]: nit 
```suggestion
apiVersion: %[1]s
```

File: cmd/kubeadm/app/util/config/common_test.go [hunk 6]
```diff
@@ -240,29 +241,29 @@ func TestMigrateOldConfigFromFile(t *testing.T) {
...
 		{
 			desc: "Init + Cluster Configurations are migrated",
-			oldCfg: dedent.Dedent(`
-			apiVersion: kubeadm.k8s.io/v1beta2
+			oldCfg: dedent.Dedent(fmt.Sprintf(`
+			apiVersion: %s
 			kind: InitConfiguration
 			---
-			apiVersion: kubeadm.k8s.io/v1beta2
+			apiVersion: %s
```
Comment (fabriziopandini) [inline cmd/kubeadm/app/util/config/common_test.go:264]: nit:
```suggestion
			apiVersion: %[1]s
```

File: cmd/kubeadm/app/util/config/common_test.go [hunk 7]
```diff
@@ -271,18 +272,18 @@ func TestMigrateOldConfigFromFile(t *testing.T) {
...
 		{
 			desc: "Init + Join Configurations are migrated",
-			oldCfg: dedent.Dedent(`
-			apiVersion: kubeadm.k8s.io/v1beta2
+			oldCfg: dedent.Dedent(fmt.Sprintf(`
+			apiVersion: %s
 			kind: InitConfiguration
 			---
-			apiVersion: kubeadm.k8s.io/v1beta2
+			apiVersion: %s
```
Comment (fabriziopandini) [inline cmd/kubeadm/app/util/config/common_test.go:279]: nit
```suggestion
			apiVersion: %[1]s
```

File: cmd/kubeadm/app/util/config/common_test.go [hunk 8]
```diff
@@ -292,18 +293,18 @@ func TestMigrateOldConfigFromFile(t *testing.T) {
...
 		{
 			desc: "Cluster + Join Configurations are migrated",
-			oldCfg: dedent.Dedent(`
-			apiVersion: kubeadm.k8s.io/v1beta2
+			oldCfg: dedent.Dedent(fmt.Sprintf(`
+			apiVersion: %s
 			kind: ClusterConfiguration
 			---
-			apiVersion: kubeadm.k8s.io/v1beta2
+			apiVersion: %s
```
Comment (fabriziopandini) [inline cmd/kubeadm/app/util/config/common_test.go:300]: nit
```suggestion
			apiVersion: %[1]s
```

File: cmd/kubeadm/app/util/config/common_test.go [hunk 9]
```diff
@@ -313,21 +314,21 @@ func TestMigrateOldConfigFromFile(t *testing.T) {
...
 		{
 			desc: "Init + Cluster + Join Configurations are migrated",
-			oldCfg: dedent.Dedent(`
-			apiVersion: kubeadm.k8s.io/v1beta2
+			oldCfg: dedent.Dedent(fmt.Sprintf(`
+			apiVersion: %s
 			kind: InitConfiguration
 			---
-			apiVersion: kubeadm.k8s.io/v1beta2
+			apiVersion: %s
```
Comment (fabriziopandini) [inline cmd/kubeadm/app/util/config/common_test.go:321]:
```suggestion
			apiVersion: %[1]s
```

File: cmd/kubeadm/app/util/config/common_test.go [hunk 10]
```diff
@@ -313,21 +314,21 @@ func TestMigrateOldConfigFromFile(t *testing.T) {
...
 			kind: ClusterConfiguration
 			---
-			apiVersion: kubeadm.k8s.io/v1beta2
+			apiVersion: %s
```
Comment (fabriziopandini) [inline cmd/kubeadm/app/util/config/common_test.go:324]:
```suggestion
			apiVersion: %[1]s
```

File: cmd/kubeadm/app/util/config/common_test.go [hunk 11]
```diff
@@ -337,14 +338,14 @@ func TestMigrateOldConfigFromFile(t *testing.T) {
...
 		{
 			desc: "component configs are not migrated",
-			oldCfg: dedent.Dedent(`
-			apiVersion: kubeadm.k8s.io/v1beta2
+			oldCfg: dedent.Dedent(fmt.Sprintf(`
+			apiVersion: %s
 			kind: InitConfiguration
 			---
-			apiVersion: kubeadm.k8s.io/v1beta2
+			apiVersion: %s
```
Comment (fabriziopandini) [inline cmd/kubeadm/app/util/config/common_test.go:345]:
```suggestion
			apiVersion: %[1]s
```

File: cmd/kubeadm/app/util/config/common_test.go [hunk 12]
```diff
@@ -337,14 +338,14 @@ func TestMigrateOldConfigFromFile(t *testing.T) {
...
 			kind: ClusterConfiguration
 			---
-			apiVersion: kubeadm.k8s.io/v1beta2
+			apiVersion: %s
```
Comment (fabriziopandini) [inline cmd/kubeadm/app/util/config/common_test.go:348]:
```suggestion
			apiVersion: %[1]s
```

**Review by fabriziopandini — COMMENTED** [review]
@wangyysde looks great!
A couple of nits to avoid repeating parameters in Printf statements, then ready to lgtm

**Comment by wangyysde** [comment]
@fabriziopandini  Thanks. I have modified the files to avoid repeating parameters in Printf statements.

**Review by SataQiu — COMMENTED** [review]
Thanks @wangyysde 
/lgtm

**Pull request merged**